Lib/test/libregrtest/setup.py

```diff
@@ -14,17 +15,24 @@
 
 
 def setup_tests(ns):
-    # Display the Python traceback on fatal errors (e.g. segfault)
-    faulthandler.enable(all_threads=True)
-
-    # Display the Python traceback on SIGALRM or SIGUSR1 signal
-    signals = []
-    if hasattr(signal, 'SIGALRM'):
-        signals.append(signal.SIGALRM)
-    if hasattr(signal, 'SIGUSR1'):
-        signals.append(signal.SIGUSR1)
-    for signum in signals:
-        faulthandler.register(signum, chain=True)
+    try:
+        stderr_fd = sys.stderr.fileno()
+    except io.UnsupportedOperation:
```

Comment:
Maybe catch ValueError? This would cover the case of closed files too. And if support the case `sys.stderr is None` we should catch AttributeError.

Reply:
ValueError: ok, I made the change. It avoids "import io", so I like it :-) Even if I'm not sure that we are supposed to support running tests with closed stderr...

AtttributeError: I'm not sure that it makes sense to run the test suite if there is no sys.stderr. Where are you supposed to write the output if there is not stdout nor stderr?